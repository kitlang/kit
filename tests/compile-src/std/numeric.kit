function main() {
    var a = 127_i8;
    var a2 = 127;
    var b = 255_u8;
    var c = 32767_i16;
    var d = 100000;
    var e = 127_u64;
    var f = 127_f32;
    var g = 100_s;

    if ++a == -128 {
        printf("a overflow\n");
    }
    if ++a2 == 128 {
        printf("no a2 overflow\n");
    }
    if ++b == 0 {
        printf("b overflow\n");
    }
    if ++c == -32768 {
        printf("c overflow\n");
    }
    if ++d == 100001 {
        printf("no d overflow\n");
    }
    if ++e == 0 {
        printf("e overflow\n");
    }
    f += 1.0;
    printf("%.1f\n", f);
    printf("%zu\n", g);
}
