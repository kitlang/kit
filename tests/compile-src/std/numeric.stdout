a overflow
no a2 overflow
b overflow
c overflow
no d overflow
128.0
100
